modules/gdscript/gdscript_parser.cpp

```diff
@@ -4087,8 +4089,13 @@ String GDScriptParser::DataType::to_string() const {
 			}
 			return native_type.operator String();
 		}
-		case ENUM:
-			return enum_type.operator String() + " (enum)";
+		case ENUM: {
+			// native_type contains either the native class defining the enum
+			// or the fully qualified class name of the script defining the enum
+			String native_str = native_type.operator String();
+			String file = native_str.get_file();
+			return (file == native_str) ? native_str : file;
```

Comment:
If I remember correctly, it depends on whether `class_name` was defined, at which point it uses that. Otherwise it's the entire filepath and you need to isolate the filename only!

Reply:
I mean, `(x == y) ? y : x` result will always be value of `x`. No?

Reply:
Yeah, passes all tests :) Thanks for noticing that!